### PR TITLE
Observable extensions

### PR DESCRIPTION
**Pull request by subzero911**
Dear MobX maintainers, I created some extensions to make a code more concise. I hope you find them useful.
1. Type aliases for primitive types.
So instead of `Observable<int>` you can now write `ObservableInt`.
But it uses the Dart 2.13 feature "type aliases", so I bumped sdk constraints in pubspec to min 2.13
2. `asObservable()` extensions for primitive types. Now you can easily convert literals to observables like:
```dart
var name = ''.asObservable(); // infers ObservableString
var counter = 0.asObservable(); // infers ObservableInt
```
3. `toggle()` method for ObservableBool. Lets you toggle the internal value of ObservableBool
```dart
var lights = true.asObservable();
lights.toggle(); // now it has a value of false
```

### CODE REVIEW

**Review by pavanpodila — CHANGES_REQUESTED** [review]
Please update the version in pubspec and bump the minor version. Also run set-version command in package.Json. additionally we should add an entry inn change log for this

**Comment by subzero911** [comment]
I bumped the version in pubspec.

But I don't know how to run a set-version command. Is it some npm thing?

**Comment by subzero911** [comment]
I runned `npm version 1.0.1`, is it what you wanted?

**Comment by subzero911** [comment]
@pavanpodila I runned `npm version 1.0.1`, is it what you wanted?

**Comment by subzero911** [comment]
okay, I see
I run `npm run set:versions` and updated a version in mobx.dart

**Review by pavanpodila — CHANGES_REQUESTED** [review]
We need to add a Changelog entry as well for the new version

**Comment by subzero911** [comment]
added

**Comment by subzero911** [comment]
@pavanpodila anything else?

File: mobx/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+
+- Type aliases for primitive types.\
+  So instead of `Observable<int>` you can now write `ObservableInt`.
+- `.asObservable()` extension for primitive types.\
+  Now you can easily convert literals to observables like:
+  ```dart
+  var name = ''.asObservable(); // infers ObservableString
+  var counter = 0.asObservable(); // infers ObservableInt
+  ```
+- `toggle()` method for ObservableBool. Lets you toggle the internal value of ObservableBool
```
Comment (pavanpodila) [inline mobx/CHANGELOG.md:11]: ObservableBool should be in back quotes

**Review by pavanpodila — CHANGES_REQUESTED** [review]

File: mobx/CHANGELOG.md (hunk 1 above)
Comment (subzero911) [reply mobx/CHANGELOG.md]: +

**Review by pavanpodila — APPROVED** [review]

**Comment by pavanpodila** [comment]
@all-contributors add @subzero911 for code

**Pull request merged**